subsys/net/ip/net_context.c

```diff
@@ -816,6 +816,24 @@ NET_CONN_CB(tcp_established)
 				     sys_get_be32(NET_TCP_HDR(pkt)->ack));
 	}
 
+//printk("%u %u\n", sys_get_be32(NET_TCP_HDR(pkt)->seq), context->tcp->send_ack);
+
+	/* Peer sent us packet we've already seen. Apparently, our ack was lost. */
+	if (sys_get_be32(NET_TCP_HDR(pkt)->seq) < context->tcp->send_ack) {
+printk("Received already seen pkt - ACKing\n");
+
+	set_appdata_values(pkt, IPPROTO_TCP);
```

Comment:
@andyross : Thanks for confirming all that. My biggest concern was that maybe after your commit I dug out, you made another commit to handle that case, which later was patched into regression, and I now just trying to reinvent the wheel (as you can see, I do the homework of reading git log, but tracing complete change history is hard, and there're indeed big code move-arounds, e.g. 29b7d9c64418bf76f3751d6240e94fe18daa52cf).

So, I'm confirming that resending the current ack seq works, but otherwise leaving the structure of this patch as it is.